pkg/reconciler/apis/apiexportendpointslice/apiexportendpointslice_reconcile.go

```diff
@@ -72,27 +77,120 @@ func (r *endpointsReconciler) reconcile(ctx context.Context, apiExportEndpointSl
 				apisv1alpha1.APIExportValid,
 				apisv1alpha1.APIExportNotFoundReason,
 				conditionsv1alpha1.ConditionSeverityError,
-				"APIExport %s|%s not found",
+				"Error getting APIExport %s|%s",
 				apiExportPath,
 				apiExportEndpointSlice.Spec.APIExport.Name,
 			)
+			conditions.MarkFalse(
+				apiExportEndpointSlice,
+				apisv1alpha1.APIExportEndpointSliceURLsReady,
+				apisv1alpha1.ErrorGeneratingURLsReason,
+				conditionsv1alpha1.ConditionSeverityError,
+				"",
+			)
+			// No need to try again
 			return nil
 		} else {
 			conditions.MarkFalse(
 				apiExportEndpointSlice,
 				apisv1alpha1.APIExportValid,
-				apisv1alpha1.InternalErrorReason,
+				reason,
 				conditionsv1alpha1.ConditionSeverityError,
 				"Error getting APIExport %s|%s",
 				apiExportPath,
 				apiExportEndpointSlice.Spec.APIExport.Name,
 			)
+			conditions.MarkUnknown(
+				apiExportEndpointSlice,
+				apisv1alpha1.APIExportEndpointSliceURLsReady,
+				apisv1alpha1.ErrorGeneratingURLsReason,
+				"",
+			)
 			return err
 		}
 	}
 	conditions.MarkTrue(apiExportEndpointSlice, apisv1alpha1.APIExportValid)
 
-	if err = r.updateEndpoints(ctx, apiExportEndpointSlice, apiExport); err != nil {
+	// Get selector from Partition
+	var selector labels.Selector
+	if apiExportEndpointSlice.Spec.Partition != "" {
+		partition, err := r.getPartition(logicalcluster.From(apiExportEndpointSlice), apiExportEndpointSlice.Spec.Partition)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Don't keep the endpoints if the Partition has been deleted
+				// and is still referenced
+				apiExportEndpointSlice.Status.APIExportEndpoints = nil
+				conditions.MarkFalse(
+					apiExportEndpointSlice,
+					apisv1alpha1.PartitionValid,
+					apisv1alpha1.PartitionInvalidReferenceReason,
+					conditionsv1alpha1.ConditionSeverityError,
+					err.Error(),
+				)
+				conditions.MarkFalse(
+					apiExportEndpointSlice,
+					apisv1alpha1.APIExportEndpointSliceURLsReady,
+					apisv1alpha1.ErrorGeneratingURLsReason,
+					conditionsv1alpha1.ConditionSeverityError,
+					"",
+				)
+				// No need to try again
+				return nil
+			} else {
+				conditions.MarkFalse(
+					apiExportEndpointSlice,
+					apisv1alpha1.PartitionValid,
+					apisv1alpha1.InternalErrorReason,
+					conditionsv1alpha1.ConditionSeverityError,
+					err.Error(),
```

Comment:
didn't you want to hide errors?

Reply:
No. I wanted to avoid leaking information and it does not. APIExportEndpointSlice and Partitions are cluster scoped resources in the same workspace.

Reply:
i am confused how does it differ from https://github.com/kcp-dev/kcp/pull/2469/files#diff-a028d37be8294707fd6fdcb63d85f41486832f61eb70c14605fc5fc712958f48R94 ?

Reply:
An APIExport from a different workspace may be referenced in the APIExportEndpointSlice. It is not the case for a Partition (only considering the same workspace). By creating an APIExportEndpointSlice (or an APIBinding) a user may get additional information on the APIExport and other workspaces if the error was returned. This scenario does not happen for Partition.

Reply:
I was thinking that an invalid request the one that would use an incorrect APIExport would be rejected by https://github.com/kcp-dev/kcp/pull/2560/files#r1064359708